bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1127,6 +1138,9 @@ void PreProcessor::finalizePreProcessing(NodeIdType clientId, uint16_t reqOffset
       // controlled by the replica while all PreProcessReply messages get released here.
       if (ReplicaConfig::instance().preExecutionResultAuthEnabled) {
         auto sigsList = reqProcessingStatePtr->getPreProcessResultSignatures();
+        // append primary's signature
+        sigsList.push_back(
```

Comment:
In case the primary signature is a part of RequestProcessingState::preProcessingResultHashes_, this is not needed.